src/store/Favorite/FavoriteProvider.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useReducer } from 'react';
+import { CartItem, FavoriteItemsContext } from '../types';
+import FavoriteContext from './favorite-context';
+
+interface FavoriteProviderProps {
+  children: React.ReactNode;
+}
+
+enum ActionType {
+  ADD = 'ADD',
+  INCREMENT = 'INCREMENT',
+  REMOVE = 'REMOVE',
+  DECREMENT = 'DECREMENT',
+}
+
+interface Action {
+  type: ActionType.DECREMENT | ActionType.REMOVE | ActionType.INCREMENT;
+  payload: Pick<CartItem, 'id'>;
+}
+
+interface ActionAdd {
+  type: ActionType.ADD;
+  payload: CartItem;
+}
+
+const favoriteReducer = (
+  state: CartItem[],
+  action: Action | ActionAdd,
+): CartItem[] => {
+  switch (action.type) {
+    case ActionType.ADD:
+      return [...state, action.payload];
+
+    case ActionType.REMOVE:
+      return state.filter(item => item.id !== action.payload.id);
+
+    case ActionType.INCREMENT:
```

Comment:
Not sure if you need `increment` and `decrement` operations for favorites

Reply:
I added it because in Favorite Items added button (ShopingButton) was the same as in ProductItem Component. It was done for adding items from the wish list to the order table. In this button there are methods increment and decrement